### PR TITLE
Fixing error message for tag

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Fixes incorrect error message for category import

Test these changes by:
- [ ] When importing a csv with categories that do not exist in the system you should receive an error message saying `The field :param1 must be a valid category id, Category id: :param2`
where :param1 is the name of the category field and :param2 is the incorrect id/tag

Fixes ushahidi/platform# .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1760)

### CODE REVIEW

File: application/messages/post.php [hunk 1]
```diff
@@ -27,6 +27,7 @@
 		'digit'         => 'The field :param1 must be a digit, Given: :param2',
 		'email'         => 'The field :param1 must be an email address, Given: :param2',
 		'exists'        => 'The field :param1 must be a valid post id, Post id: :param2',
+    'tagExists'     => 'The field :param1 must be a valid category id, Category id: :param2',
```
Comment (Angamanga) [inline application/messages/post.php]: Minor issue, the category can be identified with category name too...

File: application/classes/Ushahidi/Repository/Tag.php [hunk 2]
```diff
@@ -116,7 +116,8 @@ public function create(Entity $entity)
 	public function update(Entity $entity)
 	{
 		$tag = $entity->getChanged();
-
+		// removing children before saving tag
+		unset($tag['children']);
```
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Tag.php]: Is this necessary? I thought I made `children` part of the immutable fields for Tag already.

File: application/classes/Ushahidi/Repository/Form.php [hunk 3]
```diff
@@ -83,9 +83,12 @@ public function update(Entity $entity)
             $key === 'name' ? $this->emit($this->event, $user->email, ['primary_survey_name' => $val]) : null;
           }
         }
-
+        $form = $entity->getChanged();
+        $form['updated'] = time();
+        // removing tags from form before saving
+        unset($form['tags']);
```
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Form.php]: I already made `tags` part of the immutable fields on the object. I'm not sure why we still need to do this. I couldn't figure out how to reproduce it locally.

File: application/classes/Ushahidi/Repository/Tag.php (hunk 2 above)
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Tag.php]: Ah immutable has no effect, because the previous value was 'empty' so it just sets it :/

**Pull request merged**